core/commands/dht.go

```diff
@@ -69,9 +69,13 @@ var queryDhtCmd = &cmds.Command{
 		events := make(chan *notif.QueryEvent)
 		ctx := notif.RegisterForQueryEvents(req.Context(), events)
 
-		k := string(b58.Decode(req.Arguments()[0]))
+		id, err := peer.IDB58Decode(req.Arguments()[0])
```

Comment:
This broke a test or two but it's *strictly* more correct.

Reply:
sounds like faulty tests!

Reply:
Yeah. I just wonder if anyone was relying on this in practice. One could theoretically use this to find the peers closest to some base58 encoded thing that's *not* a valid peer ID.

Reply:
should probably make sure that this isnt how we do random-walk bootstrapping

Reply:
oh wait, this is in a command. Nvm